docs/DeveloperGuide.md

```diff
@@ -154,6 +154,47 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Sort Persons
+
+The sort persons feature allows the user to sort persons by their name, date of birth or gender. Sorting is performed as part of `listPersonsCommand#execute()`.
+
+The sorting operation is exposed in the `Model` interface as `Model#sortPersons()` which calls `AddressBook#sortPersons()` which in turn calls `UniquePersonList#sort()` to sort the underlying `ObservableList<Person>`.
+
+The following sequence diagram illustrates the relevant sorting method calls for the command `listPersons s/n` with a `SortField` object `s`:
+
+![SortPersonsSequenceDiagram](images/SortPersonsSequenceDiagram.png)
+
+
+The field to sort persons by is encapsulated within the `SortField` class. `SortField` stores the sort field enumeration (`SortFieldType`) as well as the `Comparator` object that defines the sorting logic.
+
+The following class diagram shows how the `SortField` class integrates with the other components:
+
+![SortFieldClassDiagram](images/SortFieldClassDiagram.png)
+
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** Sorting is performed directly on the underlying `UniquePersonList` object, which means the sorted result is **permanent**. i.e. Not a temporary result for the `listPersons` command. For example, if `listPersons s/n` and `listPersons` are executed back-to-back, the result of the second `listPersons` command will still display the sorted results from the first `listPersons s/n` command because the sorted result is permanent.<br><br>
+
+   Ideally, sorting should be a "view" level operation that doesn't change the underlying persons list. However, the way the UI is designed makes it difficult to implement sorting as such. The issue is that the UI is **hardcoded** to only display persons from the `UniquePersonList` object. As such, any changes to the persons list must be made directly to the `UniquePersonList` object. i.e. A modified copy of the persons list cannot be passed to the UI during runtime since the UI is hardcoded to show only the `UniquePersonList` object.<br><br>
```

Comment:
Hmmm... I would add this under Design Considerations as this issue was something that was raised up during our previous meetings, although I can understand that it would conflict with the listEvents DG because the point raised might be the same, I believe that since this is an individual DG, it would be fine to do so.